plugins/csp-wms-overlays/src/WebMapTextureLoader.cpp

```diff
@@ -0,0 +1,358 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2019 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "WebMapTextureLoader.hpp"
+
+#include "WebMapException.hpp"
+
+#include "logger.hpp"
+
+#include "../../../src/cs-utils/convert.hpp"
+#include "../../../src/cs-utils/filesystem.hpp"
+#include "../../../src/cs-utils/utils.hpp"
+
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/range/algorithm/replace_copy_if.hpp>
+#include <curlpp/Infos.hpp>
+#include <curlpp/Options.hpp>
+
+#define STB_IMAGE_IMPLEMENTATION
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb_image.h>
+#include <stb_image_write.h>
+
+namespace csp::wmsoverlays {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+WebMapTextureLoader::WebMapTextureLoader()
+    : mThreadPool(32) {
```

Comment:
Why use a fixed number of threads? Would a number relative to the number of cores be better?

Reply:
As the threads are mostly IO bound, I think the optimal amount of threads is not really dependent on the number of cores. When requesting new textures with curl I doubt the CPU will be a bottleneck, so it shouldn't be necessary to reduce the thread count on processors with fewer cores.
I agree though, that the magic number is not really an elegant solution. However, even when creating a number of threads relative to the core count I would propose running a fixed amount >1 of threads on each core, which again would introduce another magic number. What do you think about this?